coursier/src/main/scala/sbt/librarymanagement/coursier/CoursierDependencyResolution.scala

```diff
@@ -0,0 +1,259 @@
+package sbt.librarymanagement.coursier
+
+import java.io.{ File, OutputStreamWriter }
+
+import coursier.{ Artifact, Resolution, _ }
+import coursier.util.{ Gather, Task }
+import sbt.librarymanagement.Configurations.{ CompilerPlugin, Component, ScalaTool }
+import sbt.librarymanagement._
+import sbt.util.Logger
+
+case class CoursierModuleDescriptor(
+    directDependencies: Vector[ModuleID],
+    scalaModuleInfo: Option[ScalaModuleInfo],
+    moduleSettings: ModuleSettings,
+    extraInputHash: Long
+) extends ModuleDescriptor
+
+case class CoursierModuleSettings() extends ModuleSettings
+
+private[sbt] class CoursierDependencyResolution(resolvers: Vector[Resolver])
```

Comment:
Couldn't this be like a case class, or made behave like one? That way, fields corresponding to the [various sbt-coursier keys](https://github.com/coursier/sbt-coursier/blob/cbcb0761ae6c476ea86bead26cc50005ae6aff12/modules/sbt-coursier/src/main/scala/coursier/CoursierPlugin.scala#L15-L54) could correspond to fields here, that could be customized via a `.copy(…)`.

Users could then easily tweak various parameters of the resolution.

And it might also be interesting to have a next version of sbt-coursier just set and / or customize `dependencyResolution`, updating the various fields. (That would provide a migration path sbt-coursier -> lm-coursier).

Using a simple case class might pose some binary compatibility issues though… Maybe contraband could help here? (Maybe via a separate class to keep all the parameters.)

Reply:
All the keys don't necessarily need to be here in a first time, just something case-class like, fine with binary compatibility if we add fields, would allow to add the various fields later.

Reply:
I'm adding a contraband `CoursierSettings` class, this also affect previous comments around contraband itself